### PR TITLE
CI Speedup [1/3]: Use ccache to speed up CI builds

### LINKED ISSUE #501: CI build caching
### Role

Contributor

### Suggestion

The CI should cache builds with ccache/sccache/similar

### Benefit

Faster build times from CI

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### PR DESCRIPTION
**Pull request by txtsd**
Linux and macOS build very fast with this (<5mins)
Windows i686 build is down from ~40mins to ~25mins
Windows x86_64 builds in ~15mins

First build in this repo will be slow as usual. Once the ccache is uploaded, subsequent builds will be fast.

Closes #501

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -62,6 +62,32 @@ jobs:
             cmake:p
             ninja:p
             qt5:p
+            ccache:p
+
+      - name: Setup ccache
+        if: runner.os != 'Windows'
+        uses: hendrikmuhs/ccache-action@v1.2.1
+        with:
+          key: ${{ matrix.os }}-${{ matrix.appimage }}-${{ inputs.build_type }}
```
Comment (DioEgizio) [inline .github/workflows/build.yml:71]: This breaks windows ccache

Comment (DioEgizio) [reply .github/workflows/build.yml:71]:
```suggestion
          key: ${{ matrix.os }}-${{ matrix.appimage }}-${{ matrix.msystem }}-${{ inputs.build_type }}
```

**Review by DioEgizio — APPROVED** [review]

**Review by d-513 — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Review by Scrumplex — COMMENTED** [review]
Maybe we want to disable this for release builds? It's generally good practice to not rely on caches much for release builds

**Comment by txtsd** [comment]
Yea, that's a good idea. I'll do that.

**Review by timoreo22 — APPROVED** [review]

**Comment by kthchew** [comment]
Seems like release builds still try to use ccache, but since it’s not installed during release builds, it causes build failures. Strangely, there don't seem to be any failures for Windows builds.

Reproduced while testing a release on my own fork: https://github.com/kthchew/PolyMC/actions/runs/2360751464

**Comment by txtsd** [comment]
Strange. I'm not home today, but I'll fix it as soon as I'm back.

**Pull request merged**